src/Pos/Block/Logic/Internal.hs

```diff
@@ -79,17 +82,29 @@ type MonadBlockBase ssc ctx m
        -- Needed by some components.
        , MonadGState m
        -- This constraints define block components' global logic.
-       , HasLens TxpGlobalSettings ctx TxpGlobalSettings
        , HasLens LrcContext ctx LrcContext
+       , HasLens TxpGlobalSettings ctx TxpGlobalSettings
        , SscGStateClass ssc
+       , HasLens GenesisStakeholders ctx GenesisStakeholders
+       , MonadDelegation ctx m -- TODO: split dlgApplyBlocks blocks
```

Comment:
My point was:
1. Normalization while doing application goes super natural for dlg
2. Maybe there are reasons to separate them though i don't see them (might have missed the discussion).